### PR TITLE
Add property $allowArray to NumberValidator.php

### PR DESCRIPTION
**Pull request by raidkon**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | comma-separated list of tickets # fixed by the PR, if any

### CODE REVIEW

**Comment by bizley** [comment]
To be honest I'm not sure why we have this `allowArray` in `exist` and `range` validators and why we need to have it here as well. Or rather the other way around - why not have it in all the validators, what is so special in those three?

**Comment by raidkon** [comment]
@bizley 
It would be convenient if all validators had this property, but these 3 are most often (in my experience) used to transfer some kind of identifiers, and it is often necessary to be able to transfer arrays in them, it is not known whether this is used in other validators.

**Comment by samdark** [comment]
@raidkon would be great to have a fix for all validators but these alone are OK as well.

File: framework/validators/NumberValidator.php [hunk 1]
```diff
@@ -24,6 +24,11 @@
  */
 class NumberValidator extends Validator
 {
+    /**
+     * @var bool whether to allow array type attribute. Defaults to false.
```
Comment (samdark) [inline framework/validators/NumberValidator.php:28]: Need `@since 2.0.42`.

**Review by samdark — APPROVED** [review]
Please add a line for CHANGELOG. Thanks.

**Comment by samdark** [comment]
Thanks.

**Pull request merged**